partner_compassion/models/partner_compassion.py

```diff
@@ -281,6 +299,24 @@ def _compute_thankyou_letter(self):
             partner.thankyou_letter = \
                 THANKYOU_MAPPING[partner.thankyou_preference]
 
+    @api.multi
+    @api.depends("all_segments_affinity", "all_segments_affinity.affinity")
+    def _compute_prim_sec_segments(self):
+        for partner in self:
+            affinity_for_segments = sorted([(s.affinity, s) for s in partner.all_segments_affinity])[::-1]
+
+            if affinity_for_segments and len(affinity_for_segments) > 1:
+                partner.primary_segment_id = affinity_for_segments[0][1].segment_id
+                partner.secondary_segment_id = affinity_for_segments[1][1].segment_id
+
+            elif affinity_for_segments and len(affinity_for_segments) > 0:
+                partner.primary_segment_id = affinity_for_segments[0][1].segment_id
+                partner.secondary_segment_id = None
+
+            else:
+                partner.primary_segment_id = None
+                partner.secondary_segment_id = None
```

Comment:
- You can rely on the default order of recordsets to achieve the same result (see my comment in the segment affinity class).
- You can rely on python slicing to get given elements in the recordset even if the recordset is empty. It will return an empty recordset and the `.segment_id` will return `False`
```suggestion
            partner.primary_segment_id = partner.all_segments_affinity[:1].segment_id
            partner.secondary_segment_id = partner.all_segments_affinity[1:2].segment_id
```